File: Data Structures & Algorithms/add-two-numbers/submission-0.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverse (head *ListNode)*ListNode{
	if head==nil || head.Next ==nil{
		return head
	} 
	temp:=reverse(head.Next)
	head.Next.Next=head
	head.Next=nil
	return temp
}
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    n1:= ""
	n2:=""
	unreversedl1:=reverse(l1)
	unreversedl2:=reverse(l2)

	for unreversedl1 !=nil{
		n1+=strconv.Itoa(unreversedl1.Val)
		unreversedl1 =unreversedl1.Next
	}
	for unreversedl2 != nil {
		n2+=strconv.Itoa(unreversedl2.Val)
		unreversedl2 = unreversedl2.Next
	}
	num1,_ := strconv.Atoi(n1)
	num2,_ := strconv.Atoi(n2)

	num3:=num1+num2

	sumlist:=&ListNode{}

	parse:=sumlist
	n3:=strconv.Itoa(num3)

	for _,s:=range n3{
		temp:=&ListNode{}
		
		temp.Val=int(s-'0')
		parse.Next=temp
		parse=temp
	}

	finalreversedlist:=reverse(sumlist.Next)

	return finalreversedlist 

}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    int Count(ListNode*head){
        ListNode* temp = head;
        int i=0;
        while(temp->next){
            i++;
            temp=temp->next;
        }
        return i;

    }
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode*temp = head;
        if(!head || n==0){return head;}
        if(n==1){
            head=head->next;
            temp->next=nullptr;
            return head;
        }
        int len=Count(head);
        cout<<len<<endl;
        cout<<"dsf"<<endl;
        int i=1;
        if(n<=len){
         while(i<n){
            cout<<temp->val<<endl;
            temp=temp->next;
            i++;
         }
        }
        temp->next=temp->next->next;
        return head;
        
    }
};
